csharp/ql/test/library-tests/conversion/operator/Operator.ql

```diff
@@ -1,13 +1,16 @@
 import semmle.code.csharp.Conversion
 
 from Type sub, Type sup, string s1, string s2
-where convConversionOperator(sub, sup)
-  and sub != sup
-  and s1 = sub.toString()
-  and s2 = sup.toString()
-  /* Remove certain results to make the test output consistent
-   * between different versions of .NET Core. */
-  and s2 != "FormatParam"
-  and s2 != "StringOrCharArray"
-select s1, s2
-order by s1, s2
+where
+  convConversionOperator(sub, sup) and
+  sub != sup and
+  s1 = sub.toString() and
+  s2 = sup.toString() and
+  /*
+   * Remove certain results to make the test output consistent
+   * between different versions of .NET Core.
+   */
+
```

Comment:
@aschackmull : Is the linebreak here expected?

Reply:
Expected, yes, although of course there shouldn't be blank lines like this in the body of a predicate (or where-clause as in this case).  Currently the autoformatter expects `// ..` comments to apply to the following line (or the preceding line if placed on the same line at the end).  Comments of the form `/* .. */` are divided into two cases: If they are small enough that the QL author put the entire thing including the start `/*` and end `*/` markers on one line, then they are treated somewhat similarly to `// ..` comments although they may also occur in the middle of QL syntax.  If they are multi-line then they are reformatted to put the start and end markers on their own lines and the comment text is prefixed with `*` if this is missing, additionally a linebreak is added after the comment as it is expected that longer multi-line comments are mostly used at the top-level between predicates.  In this case I'd recommend changing the comment to use the `// ..` form, although I wouldn't be completely opposed to amend the autoformatter to also allow in-predicate multi-line `/* .. */` comments (i.e. recognising this and not printing the additional linebreak).  I don't think the style guide currently talks about this, but I think it would make sense to say e.g. that `// ..` are primarily for in-predicate use and multi-line `/* .. */` are primarily for top-level comments.  That is at least what is expected in the autoformatter.

Reply:
Thanks for the thorough answer, makes perfect sense.